R/plot_xctr_old.R

```diff
@@ -9,8 +9,8 @@
 #' @export
 #'
 #' @examples
-#' plot_xctr(without_financial)
-plot_xctr <- function(data) {
+#' plot_xctr_old(without_financial)
+plot_xctr_old <- function(data) {
```

Comment:
I see we're renaming an exported function. Will this break anyone's code? 

If so we need to somehow let them know. If you know who they are you may simply `@mention` them in the PR explaining the potential problems and recommending a solution. If you don't know who might be using this code then you may consider a formal deprecation process:

https://lifecycle.r-lib.org/articles/communicate.html

Reply:
Thank you Mauro, I will keep that in mind if I touch someone else's code. 
Here, I have coded this function :D. I did that because I did not want to delete right away my old function. I will delete the _old R file once I have merged this PR. 
I remember [doing that for the Sankey plot](https://github.com/2DegreesInvesting/tiltPlot/pull/20#issue-1798934548) in the tiltPlot package. I am sorry to have not explicitly stated this in my PR description ! :)

Reply:
I might have confused you. So just in case let me give you an example of what I mean. 

You code a function but other people use your code. When your code changes their code breaks. If the only user is Anne, then you can just let her know. But if there are external users (maybe the database team) you may need to consider a more gentle deprecation process. 

``` r
# BEFORE
library(tiltPlot)

plot_xctr(without_financial)
```

![](https://i.imgur.com/piWV8hE.png)<!-- -->

``` r



# NOW
# After merging this PR, this is equivalent to `library(tiltPlot)`
devtools::load_all()
#> ℹ Loading tiltPlot

# :-( 
plot_xctr(without_financial)
#> Error in plot_xctr(without_financial): could not find function "plot_xctr"

# If we have external users, it would be nice to instead throw a deprecation warning
```

<sup>Created on 2024-02-05 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>


Imagine how frustrating it would be if dplyr decided to just rename `filter()` to `filter_old()` -- virtually everyone's code would  break.

I'm not saying you should not rename it. I'm suggesting to consider the impact. If it's none or minor, then fine. Else you should go through a more gentle deprecation process as documented here:

https://lifecycle.r-lib.org/articles/communicate.html

Reply:
Oh I see!! That makes a whole lot of sense !! Thank you !